libcodechecker/analyze/log_parser.py

```diff
@@ -26,14 +28,90 @@
     ['^-m(32|64)',
      '^-std=.*'])
 
+compiler_includes_dump_file = "compiler_includes.json"
+compiler_target_dump_file = "compiler_target.json"
 
-def get_compiler_includes(compiler, lang, compile_opts, extra_opts=None):
-    """
-    Returns a list of default includes of the given compiler.
-    """
+
+def get_compiler_err(cmd):
+    '''
+    Returns the stderr of a compiler invocation as string.
+    '''
+    try:
+        proc = subprocess.Popen(shlex.split(cmd),
+                                stdin=subprocess.PIPE,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+
+        _, err = proc.communicate("")
+        return err
+    except OSError as oerr:
+        LOG.error("Error during process execution: " + cmd + '\n' +
+                  oerr.strerror + "\n")
+
+
+def parse_compiler_includes(lines):
+    '''
+    Parse the compiler include paths from a string
+    '''
     start_mark = "#include <...> search starts here:"
     end_mark = "End of search list."
 
+    include_paths = []
+
+    do_append = False
+    for line in lines.splitlines(True):
+        line = line.strip()
+        if line.startswith(end_mark):
+            break
+        if do_append:
+            # On OSX there are framework includes,
+            # where we need to strip the "(framework directory)" string.
+            # For instance:
+            # /System/Library/Frameworks (framework directory)
+            fpos = line.find("(framework directory)")
```

Comment:
`string::find(substr) == -1` can be replaced with `substr not in string`, which is more _Pythonic_.

Reply:
Ok, I agree. 
Still, this is again not new code despite it looks like that in the diff; merge conflict resolution brought this in :(